bytecomp/translobj.ml

```diff
@@ -118,7 +119,11 @@ let transl_store_label_init glob size f arg =
             Lprim (Pccall prim_makearray, [int !method_count; int 0])]),
      expr))
   in
-  (size, transl_label_init expr)
+  let lam, size = transl_label_init_general (fun () -> (expr, size)) in
+  size, lam
+
+let transl_label_init f =
+  transl_label_init_general f
```

Comment:
More precisely, turning that into a function allow to run translcore on the expressions after initializing some tables required for objects in native code.